solr/core/src/java/org/apache/solr/handler/admin/StatusOp.java

```diff
@@ -39,6 +42,14 @@ public void execute(CoreAdminHandler.CallInfo it) throws Exception {
     String reconcile = params.get(CoreAdminParams.RECONCILE_THRESHOLD);
     boolean isIndexInfoNeeded = Boolean.parseBoolean(null == indexInfo ? "true" : indexInfo);
     NamedList<Object> status = new SimpleOrderedMap<>();
+    NamedList<Object> nodeStatus = new SimpleOrderedMap<>();
+    if (it.handler.coreContainer != null) {
+      Path solrDataHome = it.handler.coreContainer.getCoreRootDirectory();
+      File dataDir = solrDataHome.toFile();
+      nodeStatus.add("indexDiskSizeTotal", dataDir.getTotalSpace());
```

Comment:
`getTotalSpace()` and `getUsableSpace()` use filesystem info and return the total and available space on the filesystem that is mounted at (or contains) the `File`. Need to test or inspect the code to see if this is effectively doing a `statfs` syscall, or doing some other expensive option.

Reply:
These calls get the info from the filesystem, and not any sort of FS walk! I've also tested for performance, and in playpen, on solr-c91-1 here are some numbers:

QTimes for https://solr-c91-1.internal.onfire.fyi/api/cores?indexInfo=true 
without node size: 115-125
with node size: 120-130

So there is a small performance impact, but only around 5ms.